plugins/jenkins_status_checker/app/models/jenkins_status_checker.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+require 'jenkins_api_client'
+require 'singleton'
+# checks jenkins staging status and publishes unstable builds checklist
+class JenkinsStatusChecker
+  include Singleton
+
+  URL = ENV['JENKINS_URL'] # 'https://jenkins.yourcompany.com'
+  USERNAME = ENV['JENKINS_USERNAME']
+  API_KEY = ENV['JENKINS_API_KEY']
+  JENKINS_CHECKLIST = ENV['JENKINS_STAGING_VIEW'] # '/view/StagingStatus'
+
+  def checklist
```

Comment:
should fail when one of the required settings are not set ... atm it fails in strange ways with nil errors

for example:
```
username: ENV.fetch('JENKINS_USERNAME'),
```